### ISSUE TITLE
Why you want backbone's gradient to multiply world_size (partial fc, pytorch version) 

### ISSUE THREAD

**Issue by Abraham423** [deepinsight/insightface#1403]
I've noticed you enforce backbone's gradient to multiply a scaler world_size , can you explain the insight behind this motivation ?

https://github.com/deepinsight/insightface/blob/79aacd2bb3323fa50a125b828bb1656166604487/recognition/partial_fc/pytorch/partial_fc.py#L190

**Comment by Abraham423** [comment]
And why you don't instead implement pytorch autograd mechanism ?

**Comment by anxiangsir** [comment]
@xiaoyang-coder

**Comment by Abraham423** [comment]
> @xiaoyang-coder

Yeah, this code is more intuitive. So whether it can converge to the same level.
 Actually, in line
https://github.com/deepinsight/insightface/blob/79aacd2bb3323fa50a125b828bb1656166604487/recognition/partial_fc/pytorch/partial_fc.py#L180
you only divide `grad`(with global length from dim0)  by a batch size from current rank rather than global batch size, that means, If i am not misunderstanding，you have magnified the `grad` value by `world_size` times. When it comes to `x_grad`, it was multiplied by `world_size`  once more,  as in line 
https://github.com/deepinsight/insightface/blob/79aacd2bb3323fa50a125b828bb1656166604487/recognition/partial_fc/pytorch/partial_fc.py#L190
  which is far bigger than the ordinary gradient, so can you share your idea behind this implement, or are you doing this just by experience for faster convergence? Thanks for your time =。=
